Vfake extra providers

ajkerrigan opened the pull request:
The `vfake` plugin is incredibly useful! One way it can become even more handy is by loading additional user-defined providers. @saulpw and I discussed this briefly, initially thinking we could use a whitespace-separated list of provider modules and pass them into the `includes` kwarg of the `Faker()` constructor. After a bit of local fiddling, I found that approach to be less flexible and a bit trickier than adding individual providers via `add_provider()`.

By allowing an option to specify a list of `Provider` classes, we open the door for `.visidatarc` configurations like this:

```python
from faker.providers import BaseProvider
from faker_cloud import AmazonWebServicesProvider
from string import ascii_uppercase, digits

class VdCustomProvider(BaseProvider):
    def ws_computer_name(self):
        return self.lexify(f"EC2AMAZ-{'?' * 7}", ascii_uppercase + digits)

options.vfake_extra_providers = [AmazonWebServicesProvider, VdCustomProvider]

Sheet.bindkey("zf", "setcol-fake")
```

In this case `vfake` has access to:

* The default Faker base providers
* AWS providers from the `faker_cloud` module (installed via `pip`)
* Some exploratory extras defined locally right inside `.visidatarc`

Custom providers will be more sustainable and shareable if they are promoted to a proper vd plugin or even a pip-installable package. Being able to define and add them right from the vd config file makes the initial customization bar much lower though, without sacrificing long-term maintainability.

ajkerrigan committed:
Allow vfake to load additional providers

After instantiating Faker, load additional providers from a user-defined
list.

saulpw commented on plugins/vfake.py:
```diff
@@ -6,11 +6,20 @@
 option('locale', 'en_US', 'default locale to use for Faker', replay=True)
 
+# See also: https://faker.readthedocs.io/en/master/communityproviders.html
+option('vfake_extra_providers', [], 'list of additional Provider classes to load via add_provider()', replay=True)
```
It's not implemented yet, but I'm thinking that having a list as option default will mean "one of these enumerated values".  Maybe change the default to `None`? (which allows any type)

ajkerrigan replied:
Ahh good tip, thanks! I changed the default, and then moved some stuff around because it was starting to make `setValuesFromFaker` look noisy.

saulpw commented:
This looks good, @ajkerrigan!  I think we should change the default (and any fallout), but otherwise I think it's good to merge.

ajkerrigan committed:
default vfake_extra_providers to None

Use None rather than an empty list as the default value - the meaning of
an empty list as an option's default value is in flux.

Break the extra provider logic out into a separate function.
